lightning/src/chain/channelmonitor.rs

```diff
@@ -1068,6 +1085,10 @@ impl<Signer: WriteableEcdsaChannelSigner> Writeable for ChannelMonitorImpl<Signe
 					upd.write(writer)?;
 				},
 				MonitorEvent::HolderForceClosed(_) => 1u8.write(writer)?,
+				// `HolderForceClosedWithInfo` replaced `HolderForceClosed` in v0.0.122. To keep
+				// backwards compatibility, we write a `HolderForceClosed` event along with the
+				// `HolderForceClosedWithInfo` event. This is deduplicated in the reader.
+				MonitorEvent::HolderForceClosedWithInfo { .. } => 1u8.write(writer)?,
```

Comment:
Alright added, think I now handle every backwards compat case :crossed_fingers:

Reply:
IIRC, the writes here are for backwards compatibility with a much older versions that predate writing the TLV record at the end. Looking at the deserialization code, we read these and then overwrite them if there is a TLV record with type `5`. Since this is an odd TLV, older versions since https://github.com/lightningdevkit/rust-lightning/commit/a515eb3ba6bf9a73db0a8a26774c83561153fb5a won't fail reading the TLV records and will get the events when reading the non-TLV section.